Stardrop/Utilities/External/NexusDownloadResult.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Stardrop.Utilities.External
+{
+    public enum DownloadResultKind
+    {
+        Failed,
+        UserCanceled,
+        Success
+    }
+
+    public record struct NexusDownloadResult(DownloadResultKind ResultKind, string? DownloadedModFilePath);
```

Comment:
One thing I don't love about this--it's possible to create an object in an invalid state. e.g. what does it mean to have `DownloadResultKind` of `Success`, but a `null` `DownloadedModFilePath`? Something something, I wish every language had Rust-style enums.